nixos/modules/security/pam.nix

```diff
@@ -410,6 +410,8 @@ let
               "password sufficient ${pam_krb5}/lib/security/pam_krb5.so use_first_pass"}
           ${optionalString config.services.samba.syncPasswordsByPam
               "password optional ${pkgs.samba}/lib/security/pam_smbpass.so nullok use_authtok try_first_pass"}
+          ${optionalString cfg.enableGnomeKeyring
+              "password optional ${pkgs.gnome3.gnome-keyring}/lib/security/pam_gnome_keyring.so use_authtok"}
```

Comment:
This is just a question out of curiosity, the previous config had a `-` in front of the management group here. Removal intentional?

Reply:
Yes, but good job spotting that!

The dash means that PAM will not log to the system if the referenced module is missing. From pam.conf(5):

```
If the type value from the list above is prepended with a - character the PAM library
will not log to the system log if it is not possible to load the module because it is
missing in the system. This can be useful especially for modules which are not
always installed on the system and are not required for correct authentication and
authorization of the login session.
```

But since positively expect the module to be present, we do want to log if it is missing.

The original line with the dash present was copypasta from Fedora's `gdm` package, were the file is not generated dynamically and the module is indeed not guaranteed to be present.

Reply:
Yes that makes perfect sense in nixos not to have the dash. 

Thanks for explaining :smile: